Update for WAFSv7.0 implementation

YaliMao-NOAA opened the pull request:
<b>Note to developers: You must use this PR template!</b>

## Description of Changes
This upgrade is for WAFS EVS, to adapt to WAFSv7.0, WAFS separation from GFSv16.

> Please include a summary of the changes and the related GitHub issue(s). Please also include relevant motivation and context.

Update data source changes of locations and filenames

## Developer Questions and Checklist
* Is this a high priorty PR? If so, why and is there a date it needs to be merged by?
* Do you have any planned upcoming annual leave/PTO?
* Are there any changes needed for when the jobs are supposed to run?
  
- [ x] The code changes follow [NCO's EE2 Standards](https://www.nco.ncep.noaa.gov/idsb/implementation_standards/ImplementationStandards.v11.0.0.pdf).
- [ ] Developer's name is removed throughout the code and have used `${USER}` where necessary throughout the code.
- [ ] References the feature branch for `HOMEevs` are removed from the code.
- [ ] J-Job environment variables, COMIN and COMOUT directories, and output follow what has been [defined](https://docs.google.com/document/d/1JWg_4q80aYmmAoD21GFjp9R9y5-3w7WGM3-0HJk0Pjs/edit#heading=h.7ysbr191vzu4) for EVS.
- [ ] Jobs over 15 minutes in runtime have restart capability.
- [ ] If applicable, changes in the `dev/drivers/scripts` or `dev/modulefiles` have been made in the corresponding `ecf/scripts` and `ecf/defs/evs-nco.def`? 
- [ ] Jobs contain the approriate file checking and don't run METplus for any missing data.
- [ ] Code is using METplus wrappers structure and not calling MET executables directly.
- [ ] Log is free of any ERRORs or WARNINGs.

## Testing Instructions

> Please include testing instructions for the PR assignee. Include all relevant input datasets needed to run the tests.
1. In dev/drivers/scripts/stats/wafs/jevs_wafs_atmos_stats.sh, add these two lines:
export DCOMINuk=/lfs/h1/ops/dev/dcom/test
export COMINwafs=/lfs/h2/emc/ptmp/yali.mao/wafsx001/prod/com/$COMPONENT/$wafs_ver
2. In versions/run.ver, uncomment the first line, "#export evs_ver=v1.0.14"
3. qsub dev/drivers/scripts/stats/wafs/jevs_wafs_atmos_stats.sh

Commit:
Update for WAFSv7.0 implementation

malloryprow commented on dev/drivers/scripts/stats/wafs/jevs_wafs_atmos_stats.sh:
I think these two lines should be removed. We can add them for testing, but when WAFS v7.0 is implemented these paths won't be valid anymore.

YaliMao-NOAA replied:
These two lines are where new data is

malloryprow replied:
Will they be there after WAFS v7.0 is implemented?

malloryprow replied:
Hi @YaliMao-NOAA, just wanted to nudge the question up again.

These two lines should be removed. These paths, I don't think, will be the paths once WAFS v7.0 is implemented. Because of the J-Job the `COMINwafs` and `DCOMINuk` have the default value of their paths in production. For testing, we can set `COMINwafs` and `DCOMINuk` in the driver to their temporary locations and they will override the default values.

PerryShafran-NOAA replied:
@YaliMao-NOAA Has this issue been resolved?  I agree with Mallory; we should remove these two lines in the dev driver, and, for testing purposes, we should be instructed how to set COMINwafs and DCOMINuk in the driver when we test.  However, we should not have these lines for COMPATH and DCOMROOT set in the driver script in this manner.  

Please remove these two lines, and then edit the instructions to tell us how to set COMINwafs and DCOMINuk when we run your tests.  I think this is the best way for us to proceed.

PerryShafran-NOAA replied:
@YaliMao-NOAA Also - in the directory where the WAFS data is located, it also shouldn't be $USER.  If this data currently resides in your directory, it has to be fixed to yali.mao, and not $USER.  

Thus, we should set

`export COMINwafs=/lfs/h2/emc/ptmp/yali.mao/wafsx001/prod/com/wafs/${wafs_ver}`

to match what you have in the J-job:

`export COMINwafs=${COMINwafs:-$(compath.py ${envir}/com/wafs/${wafs_ver})}`

Perry

YaliMao-NOAA replied:
@PerryShafran-NOAA Thank you. Since COMINwafs data comes from my folder, can my username be hard coded in the driver? Or I use $USER in my driver but write notes in the test instructions how to change the COMINwafs location?

PerryShafran-NOAA replied:
@YaliMao-NOAA Here is our general procedure for these types of things.

First, when you submit the code, neither COMINwafs nor DCOMINuk should be defined in the driver.  You have these items defined in the J-job correctly.  

However, in the *instructions ONLY*, you would inform me how to define COMINwafs and DCOMINuk.  When I test, I will place these in the driver and then run.  But these lines would not appear in the code that you submit.  You just point me to the data in the PR instructions.

Does this help?

malloryprow commented:
@YaliMao-NOAA Can you update the PR to include testing instructions?

YaliMao-NOAA commented:
@malloryprow I just updated the testing instructions. I am going to have a meeting from 11 to 1:30. I will follow up later

YaliMao-NOAA commented:
@malloryprow This PR is good to go. 
 I've compared WAFS EVS stat files from my WAFSv7.0 parallel run to the operational GFS run. The U/V/T stat files are identical but icing stat files are NOT. After I looked into details, I found the difference  comes from the difference of GCIP. To confirm my guess, I switched GCIP to the operational while keeping all other data from my parallel run, I got identical icing stat files. FYI, GCIP dumps observation data from BUFR which is accumulated continuously over time. So GCIP run at different time may be different.

Can I have a question? evs_ver in versions/run.ver is commented out which is used by WAFS dev driver. It affects COMIN which is gotten by using compath.py. May I know what's the right way to use evs_ver?

Thank you!

PerryShafran-NOAA commented:
@YaliMao-NOAA For your own runs and EVS testing purposes, the evs_ver should be uncommented.  NCO requires us to deliver with that commented out, but we need to have it uncommented when running tests.

Commit:
Remove definations of DCOMROOT and COMPATH in dev/drivers

YaliMao-NOAA commented:
@PerryShafran-NOAA I removed the two lines of COMPATH and DCOMROOT, I modified the testing instructions. Please let me know what else I should do. Thank you.

PerryShafran-NOAA commented:
@YaliMao-NOAA Thank you!  I recommend that you run your own test with this set-up, with the definitions of COMINwafs and DCOMINuk in your dev driver, so you have something to compare.

PerryShafran-NOAA commented:
@YaliMao-NOAA I completed my test.  Please note my settings in the dev driver script: /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr576test/EVS/dev/drivers/scripts/stats/wafs/jevs_wafs_atmos_stats.sh

```
#export COMPATH=/lfs/h2/emc/ptmp/yali.mao/wafsx001/prod/com/$COMPONENT
#export DCOMROOT=/lfs/h1/ops/dev/dcom/test
export COMINwafs=/lfs/h2/emc/ptmp/yali.mao/wafsx001/prod/com/wafs/${wafs_ver}
export DCOMINuk=/lfs/h1/ops/dev/dcom/test
```

The .o file is here:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr576test/EVS/dev/drivers/scripts/stats/wafs/jevs_wafs_stats.o157510501

Please check for WARNINGs, ERRORs, or other unexpected messages in the .o file.

The small stat files are here:
/lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/stats/wafs/atmos.20241006/wafs/grid2grid

The final stat files are here:
/lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/stats/wafs/wafs.20241006

Please validate the test.

YaliMao-NOAA commented:
@PerryShafran-NOAA 
The log file looks good, without errors and warnings.
The stat files are good, identical to my test run. As addressed earlier, the stat files are not identical to the operational because GCIP is different. I had another test with the operational GCIP while forecast products from WAFSv7.0 parallel run, the stat files were identical to the operational.

PerryShafran-NOAA commented:
@YaliMao-NOAA That sounds excellent.  Thanks!  I'll take a quick look at the code.

Can we also confirm that we do not need a change in the ecf file?  Can you check that as well?  

@AndrewBenjamin-NOAA Do you want to be the 2nd reviewer even though this is global verification?

AndrewBenjamin-NOAA reviewed (approved):
After looking over the code changes, and the two separate tests done for this PR, I approve these changes. Thank you Yali for all your work testing this PR

YaliMao-NOAA commented:
@PerryShafran-NOAA I must have missed the evs ecflow training. I don't know how to check ecf for WAFS EVS. Any online instructions?

PerryShafran-NOAA commented:
@YaliMao-NOAA You didn't miss ecflow training!  However, I don't think that you need ecf training to look at the ecf file, which would be found here:  https://github.com/YaliMao-NOAA/EVS/blob/wafs_separation/ecf/scripts/stats/wafs/jevs_wafs_atmos_stats.ecf

I would say that the only reason to change anything in this file would be one of the following:

1) You added a new library to the processing
2) You are changing the run time or memory (it has to match what is in the dev driver).
3) The time that the run starts is changing.

If none of the three above items are changing, then I would say that you wouldn't need to change anything in the ecf file.

HuiyaChuang-NOAA commented:
> @YaliMao-NOAA That sounds excellent. Thanks! I'll take a quick look at the code.
> 
> Can we also confirm that we do not need a change in the ecf file? Can you check that as well?
> 
> @AndrewBenjamin-NOAA Do you want to be the 2nd reviewer even though this is global verification?

@PerryShafran-NOAA I believe "ecf" was not developed by Yali, I don't believe she would be the one to check this file.  Is there someone on EVS team who is in charge of all ecf files

YaliMao-NOAA commented:
> @YaliMao-NOAA You didn't miss ecflow training! However, I don't think that you need ecf training to look at the ecf file, which would be found here: https://github.com/YaliMao-NOAA/EVS/blob/wafs_separation/ecf/scripts/stats/wafs/jevs_wafs_atmos_stats.ecf
> 
> I would say that the only reason to change anything in this file would be one of the following:
> 
> 1. You added a new library to the processing
> 2. You are changing the run time or memory (it has to match what is in the dev driver).
> 3. The time that the run starts is changing.
> 
> If none of the three above items are changing, then I would say that you wouldn't need to change anything in the ecf file.

@PerryShafran-NOAA Thank you for listing the items to check. None of them has been changed for this minor upgrade.

PerryShafran-NOAA commented:
@YaliMao-NOAA Thank you Yali,  I'll go ahead and merge the code now.  Thanks again for your hard work in developing this upgrade.

PerryShafran-NOAA commented:
@HuiyaChuang-NOAA The ecf code was developed by EIB.  We do ask if developers if there are minor changes in walltime or memory that would need to be changed in the ecf file, and to make those changes if necessary.  In this case, nothing has changed, based on Yali's comments, so we don't have to change the ecf file.  Thanks for asking!

PerryShafran-NOAA reviewed (approved):
Code works as expected.  Approved for merge.
